lib/mux/base.ex

```diff
@@ -43,7 +43,7 @@ defmodule Mux.Base do
   def new(token_id, token_secret, opts \\ []) do
     opts = Keyword.merge(@defaults, opts) |> Enum.into(%{})
 
-    Tesla.build_client([
+    Tesla.client([
```

Comment:
`Tesla.client` was introduced in 1.2. (pre-1.2 it was `Tesla.build_client`)

This library specifies dependencies as

```
{:tesla, "~> 1.0"},
```

If this code changes it will not work on pre-1.2 and therefore the dep would have to be updated. The options are:

1. Keep `Tesla.client` (anyone using Tesla 1+ can use this lib, folks using Tesla 1.2+ will see a deprecation warning)
2. Change this to `Tesla.build_client` and update dependencies to specify `tesla ~>1.2`

Given the tradeoffs, I prefer option 1 (keep this library as compatible as possible). It's slightly annoying to have the deprecation warning, but I think given the options it's better than blocking out folks who are stuck on pre-Tesla 1.2 from being able to not use this library.

For example, some folks might have an application that uses a library that might specify a Tesla dep like `"~> 1.0.0"` or `"~> 1.1.0"`. If we required `~> 1.2` they would be blocked from using `mux-elixir`.

Reply:
I reverted this change